be/dtm0_log.h

```diff
@@ -0,0 +1,235 @@
+ *  part of this group). Each entry of such a group may have many different
+ *  states, but this set of states must include the following states:
+ *
+ *	1) InProgress - the node (owner of the log) has a request.
+ *	2) Executed - one or more operations have been executed in volatile
+ *	              memory of the participant, the result of such execution
+ *	              is known and returned to the dtx user.
+ *	3) Persistent - locally persisted in transactional context.
+ *	4) Stable - sufficient number of sent operations have been “persisted”
+ *	            on the remote end which guarantees survivial of persistent
+ *	            failures.
+ *	5) DONE -  all sent operations have been “persisted” on sufficient
+ *	           number of non-failed participants.
+ *
+ *  Every participant maintains the journal record(DTM0 Log record) that
+ *  corresponds to each distributed transaction(dtx) and it can be described by
+ *  "struct m0_dtm0_log_rec" which will be stored in volatile/persistent
+ *  storage. Basically this log record will maintain txr id, group of state of
+ *  each individual participant and payload.
+ *
+ *  When a distributed transaction(dtx) is executed by participant, participant
+ *  modify its state in group of state as M0_DTPS_EXECUTED and will add/modify
+ *  DTM0 log record, for rest of the participant it will keep the state as it is
+ *  in distributed transaction(dtx).
+ *
+ *  When a distributed transaction(dtx) become persistent on participant,
+ *  participant will modify its state in group of state as M0_DTPS_PERSISTENT
+ *  and will modify DTM0 log record, for rest of the participant it will keep
+ *  the state as it is in DTM0 log record.
+ *
+ *  Originator also maintains the same distributed transaction(dtx) record in
+ *  volatile memory. Originator is expecting to get replies from participant
+ *  when distributed transaction(dtx) on participant is M0_DTPS_EXECUTED or
+ *  M0_DTPS_PERSISTENT. On originator the state of the distributed transaction
+ *  (dtx) can be M0_DTPS_INPROGRESS to M0_DTPS_PERSISTENT for each participant.
+ *
+ *  During recovery operation of any of the participant, rest of the participant
+ *  and originator will iterate over the logged journal and extract the state
+ *  of each transation for participant under recovery from logged information
+ *  and will send redo request for those distributed transaction(dtx) which  are
+ *  not M0_DTPS_PERSISTENT on participant being recovered.
+ *
+ *  Upon receiving redo request participant under recovery will log the state
+ *  of same distributed transaction(dtx) of remote participant in persistent
+ *  store.
+ *
+ * @section Usecases
+ *
+ * 1. There is a dedicated function m0_be_dtm0_log_create which will create
+ *    m0_be_dtm0_log during mkfs procedure.
+ *
+ * 2. When distributed transaction(dtx) successfully executed on participant,
+ *    DTM0 log record will be added by participant to persistent store and where
```

Comment:
"... and where ...", this sentence is ambiguous, and hard to understand.

What is this "and" connecting?
Please separate this into multiple simple sentences, and make it easy to read.

Reply:
done